go/sqlescape/ids.go

```diff
@@ -52,11 +53,61 @@ func EscapeIDs(identifiers []string) []string {
 	return result
 }
 
-// UnescapeID reverses any backticking in the input string.
-func UnescapeID(in string) string {
+// UnescapeID reverses any backticking in the input string by EscapeID.
+func UnescapeID(in string) (string, error) {
 	l := len(in)
-	if l >= 2 && in[0] == '`' && in[l-1] == '`' {
-		return in[1 : l-1]
+
+	if l == 0 {
+		return in, nil
+	}
+
+	first, last := in[0], in[l-1]
+
```

Comment:
Can further checks here be simplified by handling `l == 1` first explicitly? 

```go
	if l == 1 {
		if in[0] == `'` {
			return "", <error>
		}
		return in
	}
```

That way you then don't have to check `l >= 2` all the time in each case.

Reply:
Added the check